### PR TITLE
[hepmc] Finish the HepMC3 spackage update

### PR DESCRIPTION
**Pull request by HadrienG2**
- Update CMake requirement per latest master
- Account for Python variant, add python dependency if used
- Account for ROOT I/O variant, add ROOT dependency if used

### CODE REVIEW

File: var/spack/repos/builtin/packages/hepmc/package.py [hunk 1]
```diff
@@ -26,12 +26,24 @@ class Hepmc(CMakePackage):
     version('2.06.06', sha256='8cdff26c10783ed4248220a84a43b7e1f9b59cc2c9a29bd634d024ca469db125')
     version('2.06.05', sha256='4c411077cc97522c03b74f973264b8d9fd2b6ccec0efc7ceced2645371c73618')
 
-    depends_on('cmake@2.6:', type='build')
+    variant('python', default=False, description='Enable Python bindings')
+    variant('rootio', default=False, description='Enable ROOT I/O')
+
+    depends_on('cmake@2.8.9:', type='build')
+    # FIXME: Officially supports Python3, but the build system doesn't find it
+    depends_on('python@:2.99.99', when='+python')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hepmc/package.py]:
```suggestion
    depends_on('python@:2.99.99', when='+python', type=('build', 'run'))
```
Presumably?

File: var/spack/repos/builtin/packages/hepmc/package.py [hunk 2]
```diff
@@ -26,12 +26,24 @@ class Hepmc(CMakePackage):
...
+    conflicts('+python', when='@:3.1.99')
+    conflicts('+rootio', when='@:2.99.99')
 
     def cmake_args(self):
+        spec = self.spec
         return [
             '-Dmomentum:STRING=GEV',
             '-Dlength:STRING=MM',
+            '-DHEPMC3_ENABLE_PYTHON={0}'.format(spec.satisfies('+python')),
+            '-DHEPMC3_ENABLE_ROOTIO={0}'.format(spec.satisfies('+rootio'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hepmc/package.py]: I noticed the conflicts above. I take it that these variants aren't available for old versions of hepmc. Do these CMake flags cause any problems for old versions of hepmc?

File: var/spack/repos/builtin/packages/hepmc/package.py (hunk 1 above)
Comment (HadrienG2) [reply var/spack/repos/builtin/packages/hepmc/package.py]: Wouldn't that lead to a CMake configuration failure?

File: var/spack/repos/builtin/packages/hepmc/package.py (hunk 2 above)
Comment (HadrienG2) [reply var/spack/repos/builtin/packages/hepmc/package.py]: I don't think so, IIRC CMake will just print a warning about an unused variable and move on.

That being said, it does seem a bit strange to allow users to enable a feature which is not actually available on the target HepMC version. Is adding a conflict considered worse?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/hepmc/package.py]: Nah, adding a conflict is fine. You're essentially saying "out of the combinatorial space of possible build configurations, these particular configurations are disallowed".

File: var/spack/repos/builtin/packages/hepmc/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/hepmc/package.py]: What do you mean? Does it link to the Python libraries, or just run the binary?

Comment (HadrienG2) [reply var/spack/repos/builtin/packages/hepmc/package.py]: It builds Python bindings, so it links to the Python libraries.

**Pull request merged**